### PR TITLE
Updated all examples to latest OpenAI package with new syntax, plus other upgrades

### PR DESCRIPTION
**Pull request by sxflynn**
### Here are some of the major changes in this PR:

- upgraded the OpenAI package to 1.x which has completely new syntax
- Replaced all Create React App examples with Vite [see explanation from React team](https://github.com/reactjs/react.dev/pull/5487#issuecomment-1409720741)
- Switched to adding `.env` support in lieu of system environment variables to make it more beginner friendly along with a `.env.example` file

### Minor quality of life enhancements
- upgraded all `requirements.txt` packages to the latest versions, and tested
- added a FastAPI startup decorator in `/03_api` to help someone try out a `GET` request with one click to the browser
- added more setup info in the readme to help beginners, like setting up environment variables
- cleaned up `.gitignore` inconsistences 
- cleaned up random readmes in the various React folders

### CODE REVIEW

File: 03_api/api.py [hunk 1]
```diff
@@ -65,15 +65,21 @@ def parse_suggestions(text: str) -> list[Suggestion]:
 def create_app() -> FastAPI:
     app = FastAPI()
 
+    @app.on_event("startup")
+    def startup_event():
+        clickable_url = "http://127.0.0.1:8000/search?sound=shuddering&location=wheels"
+        print(f"Try out the API: {clickable_url}")
```
Comment (developing-human) [inline 03_api/api.py:73]: I found this useful coming back to the project after a few months away, thanks!

File: 03_api/api.py [hunk 2]
```diff
@@ -27,16 +29,14 @@ def troubleshoot_car(sound: str, location: str) -> list[Suggestion]:
...
-        messages=[
-            {"role": "system", "content": "You are a helpful assistant."},
-            {"role": "user", "content": prompt},
-        ]
-    )
+    response = client.chat.completions.create(model="gpt-3.5-turbo",
+    messages=[
+        {"role": "system", "content": "You are a helpful assistant."},
+        {"role": "user", "content": prompt},
+    ])
```
Comment (developing-human) [inline 03_api/api.py]: Formatting of this here/elsewhere is a little odd.  Would expect:
```
    response = client.chat.completions.create(
        model="gpt-3.5-turbo",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": prompt},
        ],
    )
```

File: README.md [hunk 3]
```diff
@@ -42,8 +60,8 @@ pip3 install -r requirements.txt
 
 ### Running examples
 - Change into directory for phase of project you're working with
-- Start the python backend with `uvicorn api:app --reload`
-- (For 4-6) run `npm install` then `npm start`
+- Starting with step 3, start the python backend with `uvicorn api:app --reload`
+- (For 4-6) run `npm install` then `npm run dev`
```
Comment (developing-human) [inline README.md:64]: I'd prefer the same phrasing between "Starting with step 3" and "(For 4-6)" but no preference as to which format is used.

File: 05_streaming/index.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>04_Frontend</title>
```
Comment (developing-human) [inline 05_streaming/index.html]: Very minor, but "04_Frontend" appears in a few places inside the 05 and 06 directories.  It very well may have been this way before too :D

**Review by developing-human — COMMENTED** [review]
Really appreciate your help in bringing things up to date.  I made a few minor comments.

In the future (here and on other projects), more bite sized PRs are usually easier to review (and revert, if anything goes wrong).  For example, this one may have been broken into one for each of: moving to vite, upgrading openai usage, introducing dotenv, upgrading packages, and other cleanup.

Again, thank you for helping bring things up to date.

File: 05_streaming/index.html (hunk 4 above)
Comment (sxflynn) [reply 05_streaming/index.html]: No, that was me. Fixed!

File: 03_api/api.py (hunk 2 above)
Comment (sxflynn) [reply 03_api/api.py]: Thanks, I went in and fixed all of them to make them more consistent with the original codebase.

File: 03_api/api.py (hunk 1 above)
Comment (sxflynn) [reply 03_api/api.py:73]: Glad you found it useful!

File: README.md (hunk 3 above)
Comment (sxflynn) [reply README.md:64]: If you'd prefer the original phrasing (but with `npm run dev`) then that's fine. I can remove the "Starting with step 3, " text

**Review by sxflynn — COMMENTED** [review]
Thanks for your feedback. I'll look into the best workflow for taking a whole bunch of commits and breaking them down into separate PRs, because you're right that PRs should have a single focus.

**Pull request merged**